### PR TITLE
Update data_synthesis.py

### PR DESCRIPTION
**Pull request by Abdevna**

### CODE REVIEW

**Comment by Abdevna** [comment]
Closes #353
Please check it @VictorOdede @abhigya-sodani

**Comment by abhigya-sodani** [comment]
I am a little confused how this will fix the issue? 

the error: TypeError: BaseOnsiteLLM.generate() got an unexpected keyword argument 'temperature', seems like a parameter issue in our BaseOnsiteLLM class in onesite_llm.py.

What was your diagnosis of the root of this problem, maybe that will help me understand better

File: src/llm_vm/completion/data_synthesis.py [hunk 1]
```diff
@@ -37,10 +37,10 @@ def data_synthesis(self, optimizer, prompt, response, openai_key=None, completio
             return list(pickle.load(new_file))
         datapoints = []
         final_prompt = '{"prompt": "' +prompt+'"  , "response": "' +response+'" }'
-        final_prompt = "Generate 1 json similar to the one below. \n" + final_prompt
+        final_prompt = "Generate 1 json similar to the one below. \n" + json.dumps(final_prompt)
```
Comment (abhigya-sodani) [inline src/llm_vm/completion/data_synthesis.py:40]: What is the motivation behind doing this?

Comment (Abdevna) [reply src/llm_vm/completion/data_synthesis.py:40]: I had done this to ensure that the data that is generated has a particular format which will make processesing easier

**Comment by Abdevna** [comment]
@abhigya-sodani 
I provided code modifications that assumed the temperature argument should be accepted by the generate method. I overlooked the fact that the generate method of the BaseOnsiteLLM class in onesite_llm.py doesn't accept a temperature argument. To resolve that we can modify the generate method in the BaseOnsiteLLM class to accept the temperature argument, something like this:

`def generate(self, prompt, max_len, temperature=1.0, **kwargs):`

This can, I believe ensure that the `generate` method can accept the `temperature` argument when called.

**Comment by VictorOdede** [comment]
This issue was closed by #366

**Pull request closed without merging**